### PR TITLE
[Native] Add debug config for native engine execution

### PR DESCRIPTION
**Pull request by bikramSingh91**
This patch adds the required changes to allow setting a debug query config
specific to native engine. The query config added on presto side is
`native_debug.validate_output_from_operators`

Depends on: #21039 to pull in latest changes from velox

### CODE REVIEW

**Review by amitkdutta — APPROVED** [review]
Thans @bikramSingh91 for adding the configs. Looks good to me.

**Review by mbasmanova — APPROVED** [review]
@bikramSingh91 Thanks.

Please, update commit message to return the following redundant text.

```
Test Plan:
Updated and added unit tests
```

File: presto-native-execution/presto_cpp/main/tests/QueryContextManagerTest.cpp [hunk 1]
```diff
@@ -31,16 +31,31 @@ TEST_F(QueryContextManagerTest, nativeSessionProperties) {
       .systemProperties = {
           {"native_max_spill_level", "1"},
           {"native_spill_compression_codec", "NONE"},
-          {"native_join_spill_enabled", "true"},
```
Comment (majetideepak) [inline presto-native-execution/presto_cpp/main/tests/QueryContextManagerTest.cpp:34]: Should we move this to another PR?

**Review by majetideepak — CHANGES_REQUESTED** [review]
@bikramSingh91 can the Velox update be moved to another PR?

File: presto-native-execution/presto_cpp/main/tests/QueryContextManagerTest.cpp (hunk 1 above)
Comment (bikramSingh91) [reply presto-native-execution/presto_cpp/main/tests/QueryContextManagerTest.cpp:34]: This part is just expanding the unit testing and not changing any defaults so is fairly benign to keep in this PR. Please let me know if you still feel strongly about separating this.

**Comment by bikramSingh91** [comment]
> @bikramSingh91 can the Velox advance be moved to another PR?

@majetideepak Please take a look #21039

File: presto-native-execution/presto_cpp/main/tests/QueryContextManagerTest.cpp (hunk 1 above)
Comment (majetideepak) [reply presto-native-execution/presto_cpp/main/tests/QueryContextManagerTest.cpp:34]: This is fine. Thanks for clarifying!

**Comment by bikramSingh91** [comment]
@majetideepak Updated the PR and all tests pass, can you please proceed with the merge

**Review by majetideepak — APPROVED** [review]
Thanks!
